### PR TITLE
`#![deny(unsafe_op_in_unsafe_fn)]` in sys/hermit

### PR DESCRIPTION
**Pull request by maekawatoshiki**
Partial fix of #73904.

This encloses ``unsafe`` operations in ``unsafe fn`` in ``sys/hermit``.
Some unsafe blocks are not well documented because some system-based functions lack documents.

### CODE REVIEW

**Comment by Dylan-DPC** [comment]
@joshtriplett this is ready for review

**Comment by Dylan-DPC** [comment]
maybe

r? @Mark-Simulacrum

**Comment by Mark-Simulacrum** [comment]
I think it's probably best to just leave this module as-is (perhaps adding a top-level allow at most). The hermit folks are the ones who can review safety comments in it well and for now it's not necessary.

I can accept this with the changes reverted and an allow added instead of denys.

I would guess that having the current diff available though would be helpful, so posting a link to a gist or so with the diff seems good.

Thanks!

**Comment by maekawatoshiki** [comment]
This is a link to a gist with diff: https://gist.github.com/maekawatoshiki/ccd255d2e59288917ebee1fb42ba776c

File: library/std/src/sys/hermit/thread_local_dtor.rs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #![cfg(target_thread_local)]
 #![unstable(feature = "thread_local_internals", issue = "none")]
+#![allow(unsafe_op_in_unsafe_fn)]
```
Comment (Mark-Simulacrum) [inline library/std/src/sys/hermit/thread_local_dtor.rs:3]: Could we put these in `hermit/mod.rs` instead?

Comment (maekawatoshiki) [reply library/std/src/sys/hermit/thread_local_dtor.rs:3]: > Could we put these in ``hermit/mod.rs`` instead?

I'm sorry, but I didn't understand what you said. What do "these" mean?

**Comment by crlf0710** [comment]
@maekawatoshiki Ping from triage! Any updates on this?

**Comment by maekawatoshiki** [comment]
I'm a little confused about what to do next...

**Comment by crlf0710** [comment]
Triage: @maekawatoshiki I think leaving one single `#![allow(unsafe_op_in_unsafe_fn)]` at the hermit root is enough. No need to add the others.

**Comment by maekawatoshiki** [comment]
I removed ``#![allow(unsafe_op_in_unsafe_fn)]``s expect for ``mod.rs``.

**Comment by Dylan-DPC** [comment]
@Mark-Simulacrum this is ready for review

**Pull request merged**